woocommerce/woocommerce
# [Experimental] Add block text color support to interactivity filter blocks

### Changes proposed in this Pull Request:

This PR adds support for the block "text" color setting when displaying the experimental interactivity API powered collection filter blocks. The only block not currently supported is the price filter. By default adding text color colors the bar of the range filter UX control and I have a question whether the input text should be colored. I would like to do that one as a separate follow up to keep PRs small.

To do this:

1. We allow the setting in block.json
2. On the editor side the checkbox list mode already works automatically, but for dropdown we just need to display the placeholder text as the correct color to simulate. To do this we use `@emotion/styled` directly on FormTokenField component
3. On PHP side we use the existing style attribute utils to get the colors from block props, we pass these to the `Dropdown` component. To target the placeholder due to it being a pseudo element we inject a small stylesheet when rendering. For other text in the dropdown we just apply an inline style targeting `color`.

Partially resolves #42056 (just note there I have the question about text color support for price range slider)
<img width="689" alt="Screenshot 2023-12-22 at 2 01 16 PM" src="https://github.com/woocommerce/woocommerce/assets/1281828/4b32f6fc-cbce-40cd-908f-50564e6534da">
<img width="668" alt="Screenshot 2023-12-22 at 2 01 01 PM" src="https://github.com/woocommerce/woocommerce/assets/1281828/c7b045c2-4680-40ae-8cba-8707a132e7da">
<img width="489" alt="Screenshot 2023-12-22 at 2 00 38 PM" src="https://github.com/woocommerce/woocommerce/assets/1281828/dbdcd1c1-4985-4266-84e7-639453694c8f">



### How to test the changes in this Pull Request:

1. In block editor insert collection filters block, this will auto insert the rating, attribute and stock filter blocks
2. for each of those blocks, open up the toolbar menu and choose a text color. Note that in the editor the color of the checkbox list text matches the chosen color. Now in the block settings change to dropdown mode. Also see the color reflected in the placeholder text of the dropdown
3. Now save the content and view it on the frontend. See that when opening dropdowns the placeholder and option text color reflect the one chosen.
4. Also go back to the list mode and see in the front end that lists display with the chosen color.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment

Add support for text color to experimental filter blocks

</details>
